contrib_bots/lib/poll.py

```diff
@@ -0,0 +1,105 @@
+
+
+class PollHandler(object):
+    '''
+    This plugin facilitates polling.
+    '''
+
+    def __init__(self):
+        # polls is a dict of dict. First dict corresponds to poll title,
+        # second refers to answers in poll
+        self.polls = dict()
+
+        # streams is a convenience object that keeps track of the latest
+        # question for each stream.
+        self.streams = dict()
```

Comment:
I support `@poll close` and will commence development on that ASAP.

As for `@poll archive` I don't see this as necessary. I think after a poll is closed it should be saved/archived by default and can be retrieved by `@poll result <question>`. I think that if a user desires to destroy a poll they should call something like `@poll remove <question>` (I only suggest a deletion option as I presume, in your version, if a user doesn't archive a poll it would be deleted. This would still give them deletion functionality if they so desired it (personally I don't think this is super necessary however it would be a nice to have)).

With regards for `@poll list` - again - I don't feel that it's paramount to the functionality of the bot however it's a simple feature to implement and I suppose it would be cool. I am concerned about how to present the list though. If the bot aggregates >20 polls there would be a fairly large data dump into a chat. I would suggest only displaying - say - the last 5 polls. Possibly, we could use a pagination model/specification if the user was very intent on getting older data (ie. `@poll list 100` gives the last 100 polls).

Let me know what you think. I'll implement `@poll close` and will hold off on `archive` and `list`.